packages/protocol/contracts/hekla/HeklaTierProvider.sol

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.24;
+
+import "../L1/tiers/TierProviderBase.sol";
+
+/// @title HeklaTierProvider
+/// @custom:security-contact security@taiko.xyz
+contract HeklaTierProvider is TierProviderBase {
+    /// @inheritdoc ITierProvider
+    function getTierIds() public pure override returns (uint16[] memory tiers_) {
+        tiers_ = new uint16[](5);
+        tiers_[0] = LibTiers.TIER_OPTIMISTIC;
+        tiers_[1] = LibTiers.TIER_SGX;
+        tiers_[2] = LibTiers.TIER_ZKVM_RISC0;
+        tiers_[3] = LibTiers.TIER_GUARDIAN_MINORITY;
+        tiers_[4] = LibTiers.TIER_GUARDIAN;
+    }
+
+    /// @inheritdoc ITierProvider
+    function getMinTier(uint256 _rand) public pure override returns (uint16) {
+        if (_rand % 1000 == 0) {
```

Comment:
```suggestion
        if (_rand % 1000 == 1) {
```

Reply:
Avoid occupancy of the probability of sgx.

Reply:
Since SGX is in `else if`, its ok to put `0` here?